src/main/java/com/github/fzakaria/ascii85/Ascii85.java

```diff
@@ -91,10 +91,21 @@ public static String encode(byte[] payload) {
         if (chars == null || chars.length() == 0) {
             throw new IllegalArgumentException("You must provide a non-zero length input");
         }
-        //By using five ASCII characters to represent four bytes of binary data the encoded size ¹⁄₄ is larger than the original
-        BigDecimal decodedLength = BigDecimal.valueOf(chars.length()).multiply(BigDecimal.valueOf(4))
-                .divide(BigDecimal.valueOf(5));
-        ByteBuffer bytebuff = ByteBuffer.allocate(decodedLength.intValue());
+        // Because we perform compression when encoding four bytes of zeros to a single 'z', we need
+        // to scan through the input to compute the target length, instead of just subtracting 20% of
+        // the encoded text length.
+        final int inputLength = chars.length();
+        int computedLength = 0;
```

Comment:
Since the chunk into which each occurrence of 'z' is decoded is as long as a chunk for five ordinary characters,  the computation of decoded length can be rewritten like this:
```java
int zCount = 0;                                                
for(int i = 0 ; i < inputLength; i++) {                        
    if(chars.charAt(i) == 'z') ++ zCount;                      
}                                                              
int computedLength = 4 * zCount + 4 * (inputLength - zCount) / 5;
```